transformer_qa.py

```diff
@@ -0,0 +1,118 @@
+            current_max = current_min + len(padded_match_text)
+            # Collect span beginning and end as well as match id
+            span_ranges_to_matches.append((current_min, current_max, i))
+            concatenated_matches += padded_match_text
+
+        return concatenated_matches, span_ranges_to_matches
+
+    def _filter_matches(
+        self,
+        qa_outputs: List[dict],
+        span_ranges_to_matches: List[tuple],
+        all_matches: DocumentArray,
+    ) -> DocumentArray:
+        """Get a match for each of the qa outputs"""
+        ordered_matches = []
+        for qa_output in qa_outputs:
+            span_start_pos = qa_output['start']
+            # Go through all span ranges for all matches
+            for span_range_to_match in span_ranges_to_matches:
+                span_range_min, span_range_max, match_index = span_range_to_match
+                # Check only based on span starting pos to avoid
+                # case of span spanning two matchesspan spanning two matches
+                above_span_min = span_range_min <= span_start_pos
+                below_span_max = span_start_pos < span_range_max
+                # Find match that containing text that covers span
+                if above_span_min and below_span_max:
+                    # Collect the match
+                    ordered_matches.append(all_matches[match_index])
+                    # Stop inner loop and go to next qa output
+                    break
+        return ordered_matches
+
+    @requests(on='/search')
+    def generate(self, docs, **kwargs):
+        for doc in docs:
+            concatenated_texts, span_ranges_to_matches = self._concatenate_matches(
+                doc.matches
+            )
+            qa_input = [{'context': concatenated_texts, 'question': doc.text}]
+            qa_outputs = self.model(qa_input, top_k=10)
+
+            # Get matches in order of qa outputs
+            ordered_matches = self._filter_matches(
+                qa_outputs, span_ranges_to_matches, doc.matches
+            )
+
+            # Every relevant output is matched to match it was extracted from
+            assert len(qa_outputs) == len(ordered_matches)
+
+            targets = [o['answer'] for o in qa_outputs]
+            confidence_scores = [o['score'] for o in qa_outputs]
+            uris = [m.uri for m in ordered_matches]
+            tags = [m.tags for m in ordered_matches]
+            paragraphs = [m.text for m in ordered_matches]
+
+            doc.matches.clear()
+            for (target, confidence_score, uri, tag, paragraph,) in zip(
+                targets,
+                confidence_scores,
+                uris,
+                tags,
+                paragraphs,
+            ):
+                match = Document(
+                    text=target,
+                    scores={
+                        'confidence': confidence_score,
+                    },
+                    uri=uri,
```

Comment:
uri and tag are not really a general requirement, right? They are part of our internal docsqa

Reply:
I have changed the I/O document schema as suggested. Matches now will not contain additional tags/uris.